metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/StringFunctionsTest.java

```diff
@@ -408,6 +408,23 @@ public void testAppendIfMissing() throws Exception {
 
   }
 
+  @Test
+  public void testSubstring() throws Exception {
+    Map<String, Object> variables = ImmutableMap.of("s", "apache metron");
```

Comment:
To go with type validation, tests with passing down incorrect types.

like : SUBSTRING(s,'7',9)

Reply:
Agreed, we should expect an exception in that case.